pkg/util/clusteroperator/status_test.go

```diff
@@ -184,27 +235,58 @@ func TestConditionsEqual(t *testing.T) {
 			},
 		},
 		{
-			description: "check condition message differs",
+			description: "check duplicate with single condition",
 			expected:    false,
 			a: []configv1.ClusterOperatorStatusCondition{
 				{
 					Type:    configv1.OperatorAvailable,
-					Status:  configv1.ConditionFalse,
 					Message: "foo",
 				},
 			},
 			b: []configv1.ClusterOperatorStatusCondition{
 				{
 					Type:    configv1.OperatorAvailable,
-					Status:  configv1.ConditionFalse,
-					Message: "bar",
+					Message: "foo",
+				},
+				{
+					Type:    configv1.OperatorAvailable,
+					Message: "foo",
+				},
+			},
+		},
+		{
+			description: "check duplicate with multiple conditions",
+			expected:    false,
```

Comment:
Looks like this is the one case where the old and new implementations differ.  Not sure it matters, but the new behavior seems reasonable and more consistent.